### PR TITLE
Functional/throttling reroutes

### PR DESCRIPTION
**Pull request by rtretyak**

### CODE REVIEW

File: services/src/functional-tests/kilda.properties.example [hunk 1]
```diff
@@ -15,3 +15,7 @@ neo.uri=bolt://localhost:7687
 mininet.endpoint=http://localhost:38080
 mininet-flowtool.endpoint=http://localhost:17191
 spring.profiles.active=virtual
+
+reroute.delay=10
+reroute.hardtimeout=30
+discovery.interval=2
```
Comment (abochkarev) [inline services/src/functional-tests/kilda.properties.example]: Add a new line at the end of the file.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/AutoReroutesSpec.groovy [hunk 2]
```diff
@@ -25,7 +25,10 @@ class AutoReroutesSpec extends BaseSpecification {
     @Autowired
     NorthboundService northboundService
 
-    def flowStatusTimeout = 7
+    @Value('${reroute.delay}')
```
Comment (abochkarev) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/AutoReroutesSpec.groovy]: reroute.delay or reroute.timeout? According to the variable name 'rerouteTimeout' - it should be second variant.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+import org.openkilda.testing.service.database.Database
+import org.openkilda.testing.service.northbound.NorthboundService
+import org.openkilda.testing.service.topology.TopologyEngineService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Narrative
+
+import java.util.concurrent.TimeUnit
+
+import static org.junit.Assume.assumeTrue
```
Comment (abochkarev) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy]: Usually, static imports are located right after `package some_package_name`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+    FlowHelper flowHelper
+    @Autowired
+    PathHelper pathHelper
+    @Autowired
+    NorthboundService northboundService
+    @Autowired
+    Database db
+
+    @Value('${reroute.delay}')
+    int rerouteTimeout
```
Comment (abochkarev) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy]: ditto: delay or timeout?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    @Value('${reroute.hardtimeout}')
+    int rerouteHardTimeout
+    @Value('${discovery.interval}')
+    int discoveryInterval
+
+    def setupOnce() {
+        //Sometimes Kilda will miss a discovery packet -> not issue a reroute,
```
Comment (abochkarev) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy]: Maybe "Sometimes Kilda **misses** ...."

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        // so need to allow at least twice that time before closing the window
```
Comment (abochkarev) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy]: no need space after //

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy (hunk 3 above)
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy]: Our checkstyle misses this for Groovy. Will fix here and add the correct order to checkstyle.xml as a separate patch

Comment (abochkarev) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/ThrottlingReroutesSpec.groovy]: Sure.

**Comment by rtretyak** [comment]
all above fixed

**Pull request merged**